src/models.py

```diff
@@ -172,30 +203,70 @@ def _get_size_weights(sizes, C):
     C = Variable(torch.sqrt(torch.Tensor([imsize[0] * imsize[1]])) / 2, requires_grad=False)
     if torch.cuda.is_available():
         w0 = w0.cuda()
         sigma = sigma.cuda()
         C = C.cuda()
-    return {'w0': w0, 'sigma': sigma, 'C': C}
+    return w0, sigma, C
 
 
-def mixed_dice_cross_entropy_loss(output, target, dice_weight, cross_entropy_weight, cross_entropy_loss=None, smooth=0,
+def mixed_dice_cross_entropy_loss(output, target, dice_weight=0.5, dice_loss=None,
+                                  cross_entropy_weight=0.5, cross_entropy_loss=None, smooth=0,
                                   dice_activation='softmax'):
+    """Calculate mixed Dice and Cross Entropy Loss.
+
+    Args:
+        output (torch.Tensor): Model output of shape (N x C x H x W).
+        target (torch.Tensor):
+            Target of shape (N x (1 + K) x H x W).
+            Where K is number of different weights for Cross Entropy.
+        dice_weight (float, optional): Weight of Dice loss. Defaults to 0.5.
+        dice_loss (function, optional): Dice loss function. If None multiclass_dice_loss() is being used.
+        cross_entropy_weight (float, optional): Weight of Cross Entropy loss. Defaults to 0.5.
+        cross_entropy_loss (function, optional):
+            Cross Entropy loss function.
+            If None torch.nn.CrossEntropyLoss() is being used.
+        smooth (float, optional): Smoothing factor for Dice loss. Defaults to 0.
+        dice_activation (string, optional):
+            Name of the activation function for Dice loss, softmax or sigmoid.
+            Defaults to 'softmax'.
+
+    Returns:
+        float: Loss value.
+
+    """
     dice_target = target[:, 0, :, :].long()
     cross_entropy_target = target
     if cross_entropy_loss is None:
         cross_entropy_loss = torch.nn.CrossEntropyLoss()
         cross_entropy_target = dice_target
+    if dice_loss is None:
+        dice_loss = multiclass_dice_loss
     return dice_weight * dice_loss(output, dice_target, smooth,
                                    dice_activation) + cross_entropy_weight * cross_entropy_loss(output,
                                                                                                 cross_entropy_target)
 
 
-def dice_loss(output, target, smooth=0, activation='softmax'):
+def multiclass_dice_loss(output, target, smooth=0, activation='softmax', excluded_classes=[]):
+    """Calculate Dice Loss for multiple class output.
+
+    Args:
+        output (torch.Tensor): Model output of shape (N x C x H x W).
+        target (torch.Tensor): Target of shape (N x H x W).
+        smooth (float, optional): Smoothing factor. Defaults to 0.
+        activation (string, optional): Name of the activation function, softmax or sigmoid. Defaults to 'softmax'.
+        excluded_classes (list, optional):
+            List of excluded classes numbers. Dice Loss won't be calculated
+            against these classes. Defaults to [].
```

Comment:
@taraspiotr Maybe example like (often used on background class) would show why/when to use it.